apps/condo/domains/common/components/tasks/TaskProgress.tsx

```diff
@@ -0,0 +1,271 @@
+
+const InfiniteSpinningStyle = css`
+  @keyframes rotate {
+    100% {
+        transform: rotate(1turn);
+    }
+  }
+  .ant-progress-circle {
+    animation: rotate 1.6s linear infinite;
+  }
+`
+
+type ICircularProgressProps = {
+    progress: TaskRecordProgress,
+}
+
+/**
+ * Gradient-colored circle that can be displayed in two states depending on value of a `progress` prop:
+ * 1. Known progress displaying as partially filled circle
+ * 2. Unknown progress displaying as "fully" filled spinning circle
+ * "Fully" filling is implemented as 99 percents to prevent displaying Ant `Progress` component in completed state
+ */
+export const CircularProgress = ({ progress }: ICircularProgressProps) => (
+    <Progress
+        type="circle"
+        width={22}
+        strokeWidth={14}
+        strokeColor={{
+            '0%': '#4CD174',
+            '100%': '#6DB8F2',
+        }}
+        strokeLinecap="square"
+        showInfo={false}
+        percent={progress === TASK_PROGRESS_UNKNOWN ? 99 : progress}
+        css={progress === TASK_PROGRESS_UNKNOWN ? InfiniteSpinningStyle : {}}
+    />
+)
+
+interface ITaskProgressProps {
+    task: TaskRecord
+    translations: TaskProgressTranslations
+    progress: TaskRecordProgress
+    removeTask: () => void
+}
+
+const TaskProgressDoneHolder = styled.div`
+    cursor: pointer;
+  
+    .anticon:first-child {
+      display: block;
+    }
+    .anticon:last-child {
+      display: none;
+    }
+  
+    &:hover {
+      .anticon:first-child {
+        display: none;
+      }
+      .anticon:last-child {
+        display: block;
+      }
+    }
+`
+
+const TaskProgressDoneIcon = ({ onClick }) => (
+    <TaskProgressDoneHolder onClick={onClick}>
+        <CheckIcon />
+        <CrossIcon />
+    </TaskProgressDoneHolder>
+)
+
+
+/**
+ * Displays task as an item in tasks list on panel
+ */
+export const TaskProgress = ({ task, translations, progress, removeTask }: ITaskProgressProps) => (
+    <List.Item>
+        <List.Item.Meta
+            title={
+                <Typography.Text strong>
+                    {translations.title(task)}
+                </Typography.Text>
+            }
+            description={translations.description(task)}
+        />
+        {task.status === TASK_COMPLETED_STATUS ? (
+            <TaskProgressDoneIcon onClick={removeTask}/>
+        ) : (
+            <CircularProgress progress={progress}/>
+        )}
+    </List.Item>
+)
+
+
+interface ITaskProgressTrackerProps {
+    task: TaskRecord
+    storage: ITasksStorage
+    calculateProgress: CalculateProgressFunc
+    onComplete?: OnCompleteFunc
+    translations: TaskProgressTranslations
+}
+
+// Prevents calling handle function multiple times when tracking component will be remounted.
+// Component will be remounted because recommended technique of updating existing notification by Ant is used,
+// that replaces the current instance to new one.
+const handledCompletedStatesOfTasksIds = []
```

Comment:
this would work module wide and collect data between page reloads. do we really want to store all that data like this?